sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParserUtils.scala

```diff
@@ -39,6 +41,17 @@ object ParserUtils {
     throw new ParseException(s"Operation not allowed: $message", ctx)
   }
 
+  def duplicateClausesNotAllowed(message: String, ctx: ParserRuleContext): Nothing = {
+    throw new ParseException(s"Found duplicate clauses: $message", ctx)
```

Comment:
We cannot merge these two functions to check the duplication?
e.g.,
```
  def checkDuplicateClauses[T](nodes: util.List[T], clauseName: String, ctx: ParserRuleContext): Unit = {
    if (nodes.size() > 1) {
      throw new ParseException(s"Found duplicate clauses: $clauseName", ctx)
    }
  }
```

Reply:
Sounds good to me!